app/models/services/cbd_commitment_hash.rb

```diff
@@ -0,0 +1,34 @@
+class Services::CbdCommitmentHash
+  def initialize(cbd_hash, commitment)
+    @cbd_hash = cbd_hash
+    @commitment = commitment
+  end
+
+  def call
+    cbd_action = @cbd_hash["action"]
+    country = @cbd_hash["contacts"][0]["country"].present? ? Country.find_by(iso: @cbd_hash["contacts"][0]["country"]["identifier"].upcase)
+        : Country.find_by(iso: "--")
+
+    {
+      cbd_id: @cbd_hash["_id"],
+      name: cbd_action["name"]["en"],
+      description: cbd_action["description"]["en"],
+      country_ids: [country.id],
+      committed_year: @cbd_hash["meta"]["createdOn"].to_date.year,
```

Comment:
Here and above, is it all worth using `dig` for these heavily nested data?

Reply:
changed